v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -80,69 +92,73 @@ func (extension *BackupVaultsBackupInstanceExtension) PostReconcileCheck(
 	}
 
 	protectionStatus := *backupInstance.Status.Properties.ProtectionStatus.Status
+	protectionStatus = strings.ToLower(protectionStatus)
 	log.V(Debug).Info(fmt.Sprintf("Protection Status is  %q", protectionStatus))
 
-	// We only want to continue if protectionStatus == ProtectionError.
-	if !strings.EqualFold(protectionStatus, protectionError) {
+	// Return success if the status is in a terminal state
+	if terminalStates.Contains(protectionStatus) {
 		log.V(Debug).Info("Returning PostReconcileCheckResultSuccess")
 		return next(ctx, obj, owner, resolver, armClient, log)
 	}
 
-	// call sync api only when protection status is ProtectionError and error code is usererror
-	var protectionStatusErrorCode string
-	protectionStatusErrorCode = strings.ToLower(*backupInstance.Status.Properties.ProtectionStatus.ErrorDetails.Code)
-	log.V(Debug).Info(fmt.Sprintf("Protection Error code is  %q", protectionStatusErrorCode))
+	if protectionStatus == protectionError {
+		// call sync api only when protection status is ProtectionError and error code is usererror
+		var protectionStatusErrorCode string
+		protectionStatusErrorCode = strings.ToLower(*backupInstance.Status.Properties.ProtectionStatus.ErrorDetails.Code)
+		log.V(Debug).Info(fmt.Sprintf("Protection Error code is  %q", protectionStatusErrorCode))
 
-	if protectionStatusErrorCode != "" && strings.Contains(protectionStatusErrorCode, "usererror") {
-		id, _ := genruntime.GetAndParseResourceID(backupInstance)
-		subscription := id.SubscriptionID
-		rg := id.ResourceGroupName
-		vaultName := id.Parent.Name
+		if protectionStatusErrorCode != "" && strings.Contains(protectionStatusErrorCode, "usererror") {
```

Comment:
Is the `protectionStatusErrorCode != ""` really needed here?

Reply:
Not really, removed it!